internals/overlord/servstate/manager_test.go

```diff
@@ -1753,72 +1560,363 @@ func (s *S) TestStopRunning(c *C) {
-
-func (f fakeLogManager) ServiceStarted(serviceName string, logs *servicelog.RingBuffer) {
-	// no-op
-}
-
 func (s *S) TestNoWorkingDir(c *C) {
-	// Service command should run in current directory (package directory)
-	// if "working-dir" config option not set.
+	// This test supplies its own service manager instance and
+	// therefore should not call the default setup.
 	dir := c.MkDir()
 	err := os.Mkdir(filepath.Join(dir, "layers"), 0755)
 	c.Assert(err, IsNil)
-	manager, err := servstate.NewManager(s.st, s.runner, dir, nil, nil, fakeLogManager{})
+	s.manager, err = servstate.NewManager(s.st, s.runner, dir, nil, nil, fakeLogManager{})
 	c.Assert(err, IsNil)
-	defer manager.Stop()
 
 	outputPath := filepath.Join(dir, "output")
 	layer := parseLayer(c, 0, "layer", fmt.Sprintf(`
 services:
     nowrkdir:
         override: replace
-        command: /bin/sh -c "pwd >%s; sleep %g"
-`, outputPath, shortOkayDelay.Seconds()+0.01))
-	err = manager.AppendLayer(layer)
+        command: /bin/sh -c "pwd >%s; sync; touch %s; sleep %g"
+`, outputPath, s.donePath, shortOkayDelay.Seconds()+0.01))
+	err = s.manager.AppendLayer(layer)
 	c.Assert(err, IsNil)
 
+	// Service command should run in current directory (package directory)
+	// if "working-dir" config option not set.
 	chg := s.startServices(c, []string{"nowrkdir"}, 1)
 	s.st.Lock()
 	c.Assert(chg.Err(), IsNil)
 	s.st.Unlock()
 
+	// Let's wait until we know the output file is available
+	s.waitForFileToAppear(c)
+
 	output, err := ioutil.ReadFile(outputPath)
 	c.Assert(err, IsNil)
 	c.Check(string(output), Matches, ".*/overlord/servstate\n")
 }
 
 func (s *S) TestWorkingDir(c *C) {
+	// This test supplies its own service manager instance and
+	// therefore should not call the default setup.
 	dir := c.MkDir()
 	err := os.Mkdir(filepath.Join(dir, "layers"), 0755)
 	c.Assert(err, IsNil)
-	manager, err := servstate.NewManager(s.st, s.runner, dir, nil, nil, fakeLogManager{})
+	s.manager, err = servstate.NewManager(s.st, s.runner, dir, nil, nil, fakeLogManager{})
 	c.Assert(err, IsNil)
-	defer manager.Stop()
 
 	outputPath := filepath.Join(dir, "output")
 	layer := parseLayer(c, 0, "layer", fmt.Sprintf(`
 services:
     wrkdir:
         override: replace
-        command: /bin/sh -c "pwd >%s; sleep %g"
+        command: /bin/sh -c "pwd >%s; sync; touch %s; sleep %g"
         working-dir: %s
-`, outputPath, shortOkayDelay.Seconds()+0.01, dir))
-	err = manager.AppendLayer(layer)
+`, outputPath, s.donePath, shortOkayDelay.Seconds()+0.01, dir))
+	err = s.manager.AppendLayer(layer)
 	c.Assert(err, IsNil)
 
 	chg := s.startServices(c, []string{"wrkdir"}, 1)
 	s.st.Lock()
 	c.Assert(chg.Err(), IsNil)
 	s.st.Unlock()
 
+	// Let's wait until we know the output file is available
+	s.waitForFileToAppear(c)
+
 	output, err := ioutil.ReadFile(outputPath)
 	c.Assert(err, IsNil)
 	c.Check(string(output), Equals, dir+"\n")
 }
+
+// Default service manager setup
+func (s *S) defaultServiceManagerSetup(c *C) {
```

Comment:
Two nits: 1) the comment is unnecessary as it just repeats the name, 2) I think the name would be clearer as `setupDefaultServiceManager` (a verb).

Reply:
Done. Thanks.